From cc2d701289ec3fe7b0e375e89981cde333bef989 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Sun, 27 Nov 2016 09:36:38 +0000
Subject: W32: Ensure that gtkwin32keys.h is not needed by GTK+ users

Throw away the gdkwin32keys.h-must-not-be-included-by-itself check,
include gdkwin32keys.h directly instead of including gdkwin32.h,
remove gdkwin32keys.h inclusion from gdkwin32.h

https://bugzilla.gnome.org/show_bug.cgi?id=775163
---
 gdk/win32/gdkwin32.h     | 6 ------
 gdk/win32/gdkwin32keys.h | 3 ---
 gtk/gtkimcontextsimple.c | 2 +-
 3 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/gdk/win32/gdkwin32.h b/gdk/win32/gdkwin32.h
index d7ba5a0..32777ad 100644
--- a/gdk/win32/gdkwin32.h
+++ b/gdk/win32/gdkwin32.h
@@ -35,12 +35,6 @@
 #include <windows.h>
 #include <commctrl.h>
 
-#define __GDKWIN32_H_INSIDE__
-
-#include <gdk/win32/gdkwin32keys.h>
-
-#undef __GDKWIN32_H_INSIDE__
-
 G_BEGIN_DECLS
 
 #ifdef INSIDE_GDK_WIN32
diff --git a/gdk/win32/gdkwin32keys.h b/gdk/win32/gdkwin32keys.h
index cbd1dfc..8c8aed1 100644
--- a/gdk/win32/gdkwin32keys.h
+++ b/gdk/win32/gdkwin32keys.h
@@ -18,9 +18,6 @@
 #ifndef __GDK_WIN32_KEYS_H__
 #define __GDK_WIN32_KEYS_H__
 
-#if !defined (__GDKWIN32_H_INSIDE__) && !defined (GDK_COMPILATION)
-#error "Only <gdk/gdkwin32.h> can be included directly."
-#endif
 
 #include <gdk/gdk.h>
 
diff --git a/gtk/gtkimcontextsimple.c b/gtk/gtkimcontextsimple.c
index 6d8b3bd..5c1fc04 100644
--- a/gtk/gtkimcontextsimple.c
+++ b/gtk/gtkimcontextsimple.c
@@ -32,7 +32,7 @@
 #include "gtkalias.h"
 
 #ifdef GDK_WINDOWING_WIN32
-#include <win32/gdkwin32.h>
+#include <win32/gdkwin32keys.h>
 #endif
 
 typedef struct _GtkComposeTable GtkComposeTable;
-- 
cgit v0.12

